tasks/static_quality_gates/lib/gates_lib.py

```diff
@@ -262,3 +273,350 @@ def generate_metric_reports(self, ctx, filename="static_gate_report.json", branc
+    os: str
+    max_on_wire_size: int
+    max_on_disk_size: int
+    artifact_on_disk_size: int
+    artifact_on_wire_size: int
+    artifact_path: str  # Path to the artifact to be used for the gate
+
+    def __init__(self, gate_name: str, gate_max_size_values: dict):
+        self.gate_name = gate_name
+        self.max_on_wire_size = read_byte_input(gate_max_size_values["max_on_wire_size"])
+        self.max_on_disk_size = read_byte_input(gate_max_size_values["max_on_disk_size"])
+        self._set_arch()
+
+    def _set_arch(self):
+        if "amd64" in self.gate_name:
+            self.arch = "amd64"
+        elif "arm64" in self.gate_name:
+            self.arch = "arm64"
+        elif "armhf" in self.gate_name:
+            self.arch = "armhf"
+        elif "msi" in self.gate_name:
+            # MSI packages are always amd64 (x86_64) on Windows
+            self.arch = "amd64"
+        else:
+            raise ValueError(f"Unknown arch for gate: {self.gate_name}")
+
+    def print_results(self) -> None:
+        """
+        Print the results of the gate
+        in case of success
+        """
+        from .static_quality_gates_reporter import QualityGateOutputFormatter
+
+        QualityGateOutputFormatter.print_enhanced_gate_result(
+            self.gate_name,
+            self.artifact_path,
+            self.artifact_on_wire_size,
+            self.max_on_wire_size,
+            self.artifact_on_disk_size,
+            self.max_on_disk_size,
+        )
+
+    @abstractmethod
+    def _measure_on_disk_and_on_wire_size(self, ctx: Context) -> None:
+        """
+        Measure the size of the artifact on disk and on wire
+        """
+        raise NotImplementedError("This method should be implemented by the subclass")
+
+    def check_artifact_size(self):
+        """
+        Check the size of the artifact on wire and on disk
+        against max_on_wire_size and max_on_disk_size.
+        If the artifact exceeds the maximum allowed size, raise a StaticQualityGateFailed exception.
+        """
+        error_message = ""
+        if self.artifact_on_wire_size > self.max_on_wire_size:
+            error_message += f"On wire size (compressed artifact size) {self.artifact_on_wire_size / 1024 / 1024} MB is higher than the maximum allowed {self.max_on_wire_size / 1024 / 1024} MB by the gate !\n"
+        if self.artifact_on_disk_size > self.max_on_disk_size:
+            error_message += f"On disk size (uncompressed artifact size) {self.artifact_on_disk_size / 1024 / 1024} MB is higher than the maximum allowed {self.max_on_disk_size / 1024 / 1024} MB by the gate !\n"
+        if error_message:
+            error_message = color_message(f"{self.gate_name} failed!\n" + error_message, "red")
+            raise StaticQualityGateFailed(error_message)
+
+    def execute_gate(self, ctx: Context):
+        """
+        Execute the quality gate.
+        """
+        from .static_quality_gates_reporter import QualityGateOutputFormatter
```

Comment:
I would strongly prefer to have the gate return some sort of "GateResults" object (could even be a dictionary) and that the caller would then take care of the printing.